docs/architecture.md

```diff
@@ -44,3 +44,92 @@ To achieve these goals, the basic flow of the controller is:
 This basic flow allows the Akri controller to ensure that protocol brokers and Kubernetes Services are running on all nodes exposing desired resources while respecting the limitations defined by `Configuration.capacity`.
 
 For a more in-depth understanding, see [Controller In-depth](./controller-in-depth.md).
+
+## Akri Flow - In Depth
+![Akri Flow Sequence Diagram](./media/flow-sequence-diagram.svg)
+**Note**: For the sake of this example, some content has been excluded from the Pod, Configuration and Instances shown below.
+1. Operator applies a Configuration with a capacity of 3 to the single node cluster.
+    ```yaml
+    kind: Configuration
+    metadata:
+      name: akri-<protocolA>
+    spec:
+      protocol:
+        <protocolA>:
+      brokerPodSpec:
+        containers:
+        - name: custom-broker
+        image: "ghcr.io/…"
+      # ...
+      capacity: 3
+    ```
+1. The Akri Agent sees the Configuration and discovers a leaf device using the protocol specified in the Configuration. It creates a device plugin for that leaf device and registers it with the kubelet. The Agent then creates an Instance for the discovered leaf device, listing itself as a node that can access it under `nodes`. The Akri Agent puts all the information that the broker pods will need in order to connect to the specific device under the `metadata` section of the Instance. Later, the controller will mount these as environment variables in the broker pods. Note how Instance has 3 available `deviceUsage` slots, since capacity was set to 3 and no brokers have been scheduled to the leaf device yet.
+    ```yaml
+    kind: Instance
+    metadata:
+      name: akri-<protocolA>-<hash>
+    spec:
+      configurationName: akri-<protocolA>
+      shared: true
+      deviceUsage:
+        akri-<protocolA>-<hash>-0: ""
+        akri-<protocolA>-<hash>-1: ""
+        akri-<protocolA>-<hash>-2: ""
+      metadata:
+        BROKER_ENV_VAR_1: <value>
+        BROKER_ENV_VAR_N: <value>
+      nodes:
+      - "<this-node>"
+    ```
+1. Controller reaches out to API Server for Instance changes. It sees that a new Instance has been created. It schedules a pod to one of the nodes on the Instance’s nodes list, adding the Instance’s name as a resource limit of the pod. Note that the pod is currently in pending state.
```

Comment:
this is definitely something we can add to the roadmap, though, and it goes along with our [proposal](https://github.com/kate-goldenring/akri/blob/main/docs/proposals/broker-deployment-strategies.md#exposing-resources-at-the-configuration-level) to expose resources on the configuration level so that the Kubernetes scheduler can make the scheduling decision.